syft/internal/fileresolver/path_skipper.go

```diff
@@ -93,12 +93,12 @@ func (ps pathSkipper) pathIndexVisitor(_ string, givenPath string, _ os.FileInfo
 	for _, mi := range ps.mounts {
 		conditionalPaths, ignorable := ps.ignorableMountTypes[mi.FSType]
 
-		if len(conditionalPaths) == 0 {
-			// Rule 1: ignore any path within a mount point that is of the given filesystem type unconditionally
-			if !containsPath(givenPath, mi.Mountpoint) {
-				continue
-			}
+		if !containsPath(givenPath, mi.Mountpoint) {
```

Comment:
A lot of the path handling is not entirely obvious. Could we add a comment indicating what this is doing? If I'm reading this correctly, this is something like:
```
// Rule 0: do not exclude mount points where the scan root is a subdirectory
```
... is this accurate?

Reply:
I've added a comment to clarify the behaviour.